### PR TITLE
Set CompatibilityVersion for templates

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #235 

cc @Eilon

### CODE REVIEW

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs [hunk 1]
```diff
@@ -58,44 +59,45 @@ public void ConfigureServices(IServiceCollection services)
 #if (IndividualLocalAuth)
             services.AddDbContext<IdentityDbContext>(options =>
     #if (UseLocalDB)
-                options.UseSqlServer(
-                    Configuration.GetConnectionString("DefaultConnection"),
-                    sqlOptions => sqlOptions.MigrationsAssembly("Company.WebApplication1")
-                ));
+                    options.UseSqlServer(
+                        Configuration.GetConnectionString("DefaultConnection"),
+                        sqlOptions => sqlOptions.MigrationsAssembly("Company.WebApplication1")
```
Comment (rynowak) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs]: Is this a tab? WHat's up with these whitepsace changes?

Comment (jbagga) [reply src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs]: I'll fix those

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs [hunk 2]
```diff
@@ -65,24 +66,24 @@ public void ConfigureServices(IServiceCollection services)
...
                 sharedOptions.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
                 sharedOptions.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
             })
     #if (OrganizationalAuth)
-            .AddAzureAd(options => Configuration.Bind("AzureAd", options))
-    #elseif (IndividualB2CAuth)
-            .AddAzureAdB2C(options => Configuration.Bind("AzureAdB2C", options))
+                .AddAzureAd(options => Configuration.Bind("AzureAd", options))
+    #elif (IndividualB2CAuth)
+                .AddAzureAdB2C(options => Configuration.Bind("AzureAdB2C", options))
```
Comment (jbagga) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs]: Keeping white spaces to indent consistently

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs [hunk 3]
```diff
@@ -65,24 +66,24 @@ public void ConfigureServices(IServiceCollection services)
...
+                ));
     #endif
-
             services.AddIdentity<IdentityUser, IdentityRole>(options => options.Stores.MaxLengthForKeys = 128)
                 .AddEntityFrameworkStores<IdentityDbContext>()
                 .AddDefaultUI()
                 .AddDefaultTokenProviders();
 
-#elseif (OrganizationalAuth || IndividualB2CAuth)
+#elif (OrganizationalAuth || IndividualB2CAuth)
```
Comment (Eilon) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs]: Were these just.... always wrong??

Comment (jbagga) [reply src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs]: After talking to @seancpeters we can change these to #elif going forward. @Eilon I'll keep this for dev

File: test/Templates.Test/EmptyWebTemplateTest.cs [hunk 4]
```diff
@@ -14,7 +14,8 @@ public EmptyWebTemplateTest(ITestOutputHelper output) : base(output)
         }
 
         [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
```
Comment (jbagga) [inline test/Templates.Test/EmptyWebTemplateTest.cs]: Added `OSSkipCondition`s cc @Eilon

**Comment by jbagga** [comment]
Linux and Windows tests pass. Mac fails because of a known error https://github.com/dotnet/corefx/issues/9728 with the following:
```
[xUnit.net 00:02:21.6105210]     Templates.Test.MvcTemplateTest.MvcTemplate_NoAuth_Works_NetCore(languageOverride: null) [FAIL]
[xUnit.net 00:02:21.6107660]       System.AggregateException : One or more errors occurred. (The handler does not support custom handling of certificates with this combination of libcurl (7.54.0) and its SSL backend ("SecureTransport").)
[xUnit.net 00:02:21.6110290]       ---- System.PlatformNotSupportedException : The handler does not support custom handling of certificates with this combination of libcurl (7.54.0) and its SSL backend ("SecureTransport").
[xUnit.net 00:02:21.6113020]       Stack Trace:
[xUnit.net 00:02:21.6114450]            at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
[xUnit.net 00:02:21.6115740]         /_/test/Templates.Test/Helpers/AspNetProcess.cs(123,0): at Templates.Test.Helpers.AspNetProcess.AssertStatusCode(String requestUrl, HttpStatusCode statusCode, String acceptContentType)
[xUnit.net 00:02:21.6118610]         /_/test/Templates.Test/MvcTemplateTest.cs(52,0): at Templates.Test.MvcTemplateTest.MvcTemplate_NoAuthImpl(String targetFrameworkOverride, String languageOverride)
[xUnit.net 00:02:21.6121030]         ----- Inner Stack Trace -----
[xUnit.net 00:02:21.6123760]            at System.Net.Http.CurlHandler.SslProvider.SetSslOptions(EasyRequest easy, ClientCertificateOption clientCertOption)
[xUnit.net 00:02:21.6125770]            at System.Net.Http.CurlHandler.EasyRequest.InitializeCurl()
[xUnit.net 00:02:21.6127450]            at System.Net.Http.CurlHandler.MultiAgent.ActivateNewRequest(EasyRequest easy)
[xUnit.net 00:02:21.6128400]         --- End of stack trace from previous location where exception was thrown ---
[xUnit.net 00:02:21.6129390]            at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__62.MoveNext()
```

I'll merge this for now but will file an issue to investigate the Mac error

**Pull request merged**